tests/components/deconz/test_device_trigger.py

```diff
@@ -135,14 +130,299 @@ async def test_get_triggers(hass, aioclient_mock):
+                    "alert": "none",
+                    "group": "10",
+                    "on": True,
+                    "reachable": True,
+                },
+                "ep": 1,
+                "etag": "1b355c0b6d2af28febd7ca9165881952",
+                "manufacturername": "IKEA of Sweden",
+                "mode": 1,
+                "modelid": "Unsupported model",
+                "name": "TRÅDFRI on/off switch ",
+                "state": {"buttonevent": 2002, "lastupdated": "2019-09-07T07:39:39"},
+                "swversion": "1.4.018",
+                "type": "ZHASwitch",
+                "uniqueid": "d0:cf:5e:ff:fe:71:a4:3a-01-1000",
+            }
+        }
+    }
+    with patch.dict(DECONZ_WEB_REQUEST, data):
+        await setup_deconz_integration(hass, aioclient_mock)
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get_device(
+        identifiers={(DECONZ_DOMAIN, "d0:cf:5e:ff:fe:71:a4:3a")}
+    )
+
+    assert device_trigger._get_deconz_event_from_device_id(hass, device.id)
+
+    triggers = await async_get_device_automations(hass, "trigger", device.id)
+
+    expected_triggers = []
+
+    assert_lists_same(triggers, expected_triggers)
+
+
+async def test_functional_device_trigger(
+    hass, aioclient_mock, mock_deconz_websocket, automation_calls
+):
+    """Test proper matching and attachment of device trigger automation."""
+    await async_setup_component(hass, "persistent_notification", {})
+
+    data = {
+        "sensors": {
+            "1": {
+                "config": {
+                    "alert": "none",
+                    "battery": 60,
+                    "group": "10",
+                    "on": True,
+                    "reachable": True,
+                },
+                "ep": 1,
+                "etag": "1b355c0b6d2af28febd7ca9165881952",
+                "manufacturername": "IKEA of Sweden",
+                "mode": 1,
+                "modelid": "TRADFRI on/off switch",
+                "name": "TRÅDFRI on/off switch ",
+                "state": {"buttonevent": 2002, "lastupdated": "2019-09-07T07:39:39"},
+                "swversion": "1.4.018",
+                "type": "ZHASwitch",
+                "uniqueid": "d0:cf:5e:ff:fe:71:a4:3a-01-1000",
+            }
+        }
+    }
+    with patch.dict(DECONZ_WEB_REQUEST, data):
+        await setup_deconz_integration(hass, aioclient_mock)
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get_device(
+        identifiers={(DECONZ_DOMAIN, "d0:cf:5e:ff:fe:71:a4:3a")}
+    )
+
+    trigger_config = {
+        CONF_PLATFORM: "device",
+        CONF_DOMAIN: DECONZ_DOMAIN,
+        CONF_DEVICE_ID: device.id,
+        CONF_TYPE: device_trigger.CONF_SHORT_PRESS,
+        CONF_SUBTYPE: device_trigger.CONF_TURN_ON,
+    }
+
+    assert await device_trigger.async_validate_trigger_config(hass, trigger_config)
```

Comment:
When does the validation happen in the normal flow of the device trigger?

Reply:
This happens in a separate task when automations are set up so just to be consistent between the different validate_trigger tests I kept a positive one. In the other tests it is not possible to assert that an exception is raised since it is running in its own task

Reply:
Won't the automation fail to be set up if the validation fails? Can we assert that?

Reply:
It probably will

Reply:
Fixed